projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-flag/components/select-flag-location/select-flag-location.component.ts

```diff
@@ -24,24 +25,27 @@ export class SelectFlagLocationComponent implements OnInit {
   private readonly caseLevelCaseFlagsFieldId = 'caseFlags';
 
   public ngOnInit(): void {
-    this.flagLocationTitle = CaseFlagWizardStepTitle.SELECT_FLAG_LOCATION;
+    this.flagLocationTitle = this.isDisplayContextParameterExternal ?
+      CaseFlagWizardStepTitle.SELECT_FLAG_LOCATION_EXTERNAL : CaseFlagWizardStepTitle.SELECT_FLAG_LOCATION;
 
     // Filter out any flags instances that don't have a party name, unless the instance is for case-level flags (this
     // is expected not to have a party name)
     if (this.flagsData) {
       this.filteredFlagsData =
-        this.flagsData.filter(f => f.flags.partyName != null || f.pathToFlagsFormGroup === this.caseLevelCaseFlagsFieldId);
+        this.flagsData.filter(f => f.flags.partyName != null || f?.pathToFlagsFormGroup === this.caseLevelCaseFlagsFieldId);
     }
-
     // Add a FormControl for the selected flag location if there is at least one flags instance remaining after filtering
     if (this.filteredFlagsData && this.filteredFlagsData.length > 0) {
       const formControl = this.formGroup.get(this.selectedLocationControlName);
+
       if (!formControl) {
         this.formGroup.addControl(this.selectedLocationControlName, new FormControl(null));
       } else {
         // Needs to be setValue as they have different object references -- we use the pathToFlagsFormGroup key
         formControl.setValue(
-          this.filteredFlagsData.find(item => item.pathToFlagsFormGroup === formControl.value.pathToFlagsFormGroup)
+          this.filteredFlagsData.find(item => {
```

Comment:
@MateiHMCTS - this can be simplified by removing the curly braces and `return` keyword, since it is just one line in the function body